test/powershell/Modules/PSDesiredStateConfiguration/PSDesiredStateConfiguration.Tests.ps1

```diff
@@ -0,0 +1,501 @@
+            Install-ModuleIfMissing -Name PSDscResources
+            $testCases = @(
+                @{
+                    TestCaseName = 'case mismatch in resource name'
+                    Name = 'groupset'
+                    ModuleName = 'PSDscResources'
+                }
+                @{
+                    TestCaseName = 'Both names have matching case'
+                    Name = 'GroupSet'
+                    ModuleName = 'PSDscResources'
+                }
+                @{
+                    TestCaseName = 'case mismatch in module name'
+                    Name = 'GroupSet'
+                    ModuleName = 'psdscResources'
+                }
+            )
+        }
+        AfterAll {
+            $Global:ProgressPreference = $origProgress
+        }
+        it "should be able to get <Name> - <TestCaseName>" -TestCases $testCases {
+            param($Name)
+
+            if($IsWindows)
+            {
+                Set-ItResult -Pending -Because "Will only find script from PSDesiredStateConfiguration without modulename"
+            }
+
+            if($IsLinux)
+            {
+                Set-ItResult -Pending -Because "https://github.com/PowerShell/PSDesiredStateConfiguration/issues/26"
+            }
+
+            $resource = Get-DscResource -Name $name
+            $resource | Should -Not -BeNullOrEmpty
+            $resource.Name | Should -Be $Name
+            if (Test-IsInvokeDscResourceEnable)
+            {
+                $resource.ImplementationDetail | Should -BeNullOrEmpty
+            }            else
+            {
+                $resource.ImplementationDetail | Should -BeNullOrEmpty
+            }
+
+        }
+
+        it "should be able to get <Name> from <ModuleName> - <TestCaseName>" -TestCases $testCases {
+            param($Name,$ModuleName, $PendingBecause)
+
+            if($IsLinux)
+            {
+                Set-ItResult -Pending -Because "https://github.com/PowerShell/PSDesiredStateConfiguration/issues/26"
+            }
+
+            if($PendingBecause)
+            {
+                Set-ItResult -Pending -Because $PendingBecause
+            }
+            $resource = Get-DscResource -Name $Name -Module $ModuleName
+            $resource | Should -Not -BeNullOrEmpty
+            $resource.Name | Should -Be $Name
+            if (Test-IsInvokeDscResourceEnable)
+            {
+                $resource.ImplementationDetail | Should -BeNullOrEmpty
+            }
+            else
+            {
+                $resource.ImplementationDetail | Should -BeNullOrEmpty
+            }
+        }
+    }
+    Context "Get-DscResource - ScriptResources" {
+        BeforeAll {
+            $origProgress = $global:ProgressPreference
+            $global:ProgressPreference = 'SilentlyContinue'
+
+            Install-ModuleIfMissing -Name PSDscResources -Force
+
+            Install-ModuleIfMissing -Name PowerShellGet -MinimumVersion '2.2.1'
```

Comment:
We include PSGet, is there a case where it isn't there so you have to install it?

Reply:
These are tests from another repo and the version of PSGet is not up to date.